backend/wayland/seat.c

```diff
@@ -563,6 +652,28 @@ void create_wl_keyboard(struct wl_keyboard *wl_keyboard, struct wlr_wl_backend *
 	wlr_signal_emit_safe(&wl->backend.events.new_input, wlr_dev);
 }
 
+void create_wl_touch(struct wl_touch *wl_touch, struct wlr_wl_backend *wl) {
+	struct wlr_wl_input_device *dev =
+		create_wl_input_device(wl, WLR_INPUT_DEVICE_TOUCH);
+	if (!dev) {
+		return;
+	}
+
+	struct wlr_input_device *wlr_dev = &dev->wlr_input_device;
+
+	wlr_dev->touch = calloc(1, sizeof(*wlr_dev->touch));
+	if (!wlr_dev->touch) {
+		wlr_log_errno(WLR_ERROR, "Allocation failed");
+		free(dev);
```

Comment:
`create_wl_input_device` does a bit more than allocation. In particular it inserts device in `backend->devices` and freeing here without removing there will lead to crash.
```suggestion
		wlr_input_device_destroy(dev);
```